plugins/amazonq/chat/jetbrains-community/tst/software/aws/toolkits/jetbrains/services/amazonqFeatureDev/controller/FeatureDevControllerTest.kt

```diff
@@ -96,12 +153,248 @@ class FeatureDevControllerTest : FeatureDevTestBase() {
+                    FollowUp(FollowUpTypes.CLOSE_SESSION, message("amazonqFeatureDev.follow_up.close_session"), status = FollowUpStatusType.Info)
+                )
+            )
+            messenger.sendChatInputEnabledMessage(testTabId, true)
+            messenger.sendUpdatePlaceholder(testTabId, message("amazonqFeatureDev.placeholder.additional_improvements"))
+        }
+    }
+
+    @Test
+    fun `test handleChat onCodeGeneration succeeds to create files`() = runTest {
+        val mockInteraction = mock<Interaction>()
+
+        val mockSession = mock<Session>()
+        whenever(mockSession.send(userMessage)).thenReturn(mockInteraction)
+        whenever(mockSession.sessionState).thenReturn(
+            PrepareCodeGenerationState(
+                testTabId, "", mock(), newFileContents, deletedFiles, references, testUploadId, 0, messenger
+            )
+        )
+
+        controller.onCodeGeneration(mockSession, userMessage, testTabId)
+
+        coVerifyOrder {
+            messenger.sendAsyncEventProgress(testTabId, true, message("amazonqFeatureDev.chat_message.start_code_generation"))
+            messenger.sendAnswer(testTabId, message("amazonqFeatureDev.chat_message.requesting_changes"), FeatureDevMessageType.AnswerStream)
+            messenger.sendUpdatePlaceholder(testTabId, message("amazonqFeatureDev.placeholder.generating_code"))
+            messenger.sendCodeResult(testTabId, testUploadId, newFileContents, deletedFiles, references)
+            messenger.sendSystemPrompt(testTabId, getFollowUpOptions(SessionStatePhase.CODEGEN, true))
+            messenger.sendUpdatePlaceholder(testTabId, message("amazonqFeatureDev.placeholder.after_code_generation"))
+            messenger.sendAsyncEventProgress(testTabId, false)
+            messenger.sendChatInputEnabledMessage(testTabId, false)
+        }
+    }
+
+    @Test(expected = RuntimeException::class)
+    fun `test handleChat onCodeGeneration throws error when sending message to state`() = runTest {
+        val mockSession = mock<Session>()
+
+        whenever(mockSession.send(userMessage)).thenThrow(RuntimeException())
+
+        controller.onCodeGeneration(mockSession, userMessage, testTabId)
+
+        coVerifyOrder {
+            messenger.sendAsyncEventProgress(testTabId, true, message("amazonqFeatureDev.chat_message.start_code_generation"))
+            messenger.sendAnswer(testTabId, message("amazonqFeatureDev.chat_message.requesting_changes"), FeatureDevMessageType.AnswerStream)
+            messenger.sendUpdatePlaceholder(testTabId, message("amazonqFeatureDev.placeholder.generating_code"))
+            messenger.sendAsyncEventProgress(testTabId, false)
+            messenger.sendChatInputEnabledMessage(testTabId, false)
+        }
+    }
+
+    @Test
+    fun `test handleChat onCodeGeneration doesn't return any files with retries`() = runTest {
```

Comment:
This test is only checking order of `messenger` methods invoked; update the test name accordingly.

Reply:
checking the messenger is in fact what the title is suggesting